java/vector/src/test/java/org/apache/arrow/vector/TestDecimalVector.java

```diff
@@ -55,8 +55,7 @@ public void test() {
     for (int i = 0; i < intValues.length; i++) {
       BigDecimal decimal = new BigDecimal(BigInteger.valueOf(intValues[i]), scale);
       values[i] = decimal;
-      decimalVector.getMutator().setIndexDefined(i);
-      DecimalUtility.writeBigDecimalToArrowBuf(decimal, decimalVector.getBuffer(), i);
+      decimalVector.getMutator().setSafe(i, decimal);
```

Comment:
Can you add a test case to cover precision/scale mismatch with the setSafe(int, BigDecimal) method?

Reply:
What mismatch to you mean exactly - like if the `BigDecimal` precision is more than Arrow supports?

Reply:
I think,

For scale, we should check BigDecimal.scale() equals the vector.scale

For precision, we should check BigDecimal.precision() <= vector.precision

Reply:
Ah, right definitely.  I'll add that thanks!